apps/src/aichat/views/modelCustomization/PublishNotes.tsx

```diff
@@ -54,52 +52,50 @@ const PublishNotes: React.FunctionComponent = () => {
     animationType: 'spin',
   };
 
+  const [alertText, type]: [string, AlertProps['type']] = hasFilledOutModelCard
+    ? ['Ready to publish', 'success']
+    : ['In order to publish, you must fill out a model card', 'warning'];
```

Comment:
Noting that this means we lose the previous formatting (bolding "must") - I'm going to open a separate task to look into the possibility of generally formatting text in the Alert component since this will affect the chat window too (spoke to Denys briefly about this).

Reply:
Just suggesting here (I don't know if the ticket isn't filed yet or my jira-fu just sucks), but my first guess at the easiest way to do this would just be to add a new attribute to `Alert` - (arbitrarily named) `Content`, and have _that_ just be any old react component.

Any attempts to bake in formatting to a component like this will tend to turn into a rat's nest of weird special cases and progressively obscure syntax as we attempt to handle all the weird cases that we'll inevitably come up with once the functionality exists. But by just handing in a component, we can kick all the formatting questions externally. So then this code could be something like this:

```
const [AlertContent, type]: [string, AlertProps['type']] = hasFilledOutModelCard
    ? [<span>Ready to publish</span?, 'success']
    : [<span>In order to publish, you <StrongText>must</StrongText> fill out a model card</span>, 'warning'];
// honestly, I don't remember offhand if we can embed jsx in this manner or if it'd need to be returned from a helper function/component, but I'll leave that as an exercise for the reader. ;-)

// and then
{!isReadOnly && <Alert Content={AlertContent} type={type} size="s" />}

// and finally, in Alert.tsx, given a new prop of "Content"
// change line 104 to this:
{ Content ? <Content/> : <span className={moduleStyles.alertText}>{text}</span> }
// which will use the new Content prop if it exists, or fall back on text if it doesn't.
```

Reply:
Yep, this makes sense to me! And nope I hadn't filed the ticket yet :) Curious what @levadadenys thinks about allowing arbitrary content into the alert (don't think it should be an issue, but I'm wondering if the design team has opinions otherwise).

Reply:
Yeah, agree, if adding support of custom html content - it can be implemented in a way @thomasoniii suggested.

However, I've spoken with design team last week and they said that `Alert` is already important/urgent enough to keep it as is, meaning we're not planning to support any bold text or custom html content in `Alerts` for now, sorry for not sharing it earlier, thought it was already discussed with everyone.

If it's crutial, might be worth discussing with design team (@markabarrett) once again.